src/main/kotlin/net/atos/zac/app/zaak/ZaakRestService.kt

```diff
@@ -809,90 +805,55 @@ class ZaakRestService @Inject constructor(
 
     @POST
     @Path("besluit")
-    fun createBesluit(@Valid besluitToevoegenGegevens: RestBesluitVastleggenGegevens): RestBesluit {
-        val zaak = zrcClientService.readZaak(besluitToevoegenGegevens.zaakUuid)
-        val zaaktype = ztcClientService.readZaaktype(zaak.zaaktype)
-        assertPolicy(policyService.readZaakRechten(zaak, zaaktype).vastleggenBesluit)
-        assertPolicy(CollectionUtils.isNotEmpty(zaaktype.besluittypen))
+    fun createBesluit(@Valid besluitToevoegenGegevens: RestBesluitVastleggenGegevens) =
+        zrcClientService.readZaak(besluitToevoegenGegevens.zaakUuid).let { zaak ->
+            ztcClientService.readZaaktype(zaak.zaaktype).let { zaaktype ->
+                assertPolicy(policyService.readZaakRechten(zaak, zaaktype).vastleggenBesluit)
+                assertPolicy(CollectionUtils.isNotEmpty(zaaktype.besluittypen))
+            }
 
-        val besluit = restBesluitConverter.convertToBesluit(zaak, besluitToevoegenGegevens)
-        zaak.resultaat?.let {
-            zgwApiService.updateResultaatForZaak(zaak, besluitToevoegenGegevens.resultaattypeUuid, null)
-        } ?: run {
-            zgwApiService.createResultaatForZaak(zaak, besluitToevoegenGegevens.resultaattypeUuid, null)
-        }
-        val restBesluit = brcClientService.createBesluit(besluit).let { restBesluitConverter.convertToRestBesluit(it) }
-        besluitToevoegenGegevens.informatieobjecten?.forEach { informatieobjectUuid ->
-            drcClientService.readEnkelvoudigInformatieobject(informatieobjectUuid).let { informatieobject ->
-                BesluitInformatieObject().apply {
-                    this.informatieobject = informatieobject.url
-                    this.besluit = restBesluit.url
-                }.let {
-                    brcClientService.createBesluitInformatieobject(
-                        it,
-                        AANMAKEN_BESLUIT_TOELICHTING
-                    )
+            besluitService.createBesluit(zaak, besluitToevoegenGegevens).let {
+                restBesluitConverter.convertToRestBesluit(it).also {
+                    // This event should result from a ZAAKBESLUIT CREATED notification on the ZAKEN channel
+                    // but open_zaak does not send that one, so emulate it here.
```

Comment:
is this something we should look into / can discuss with open_zaak?

Reply:
we can raise it as a bug in their GitHub, see how they respond...

Reply:
@edgarvonk did you remember raising this with OpenZaak?

Reply:
https://github.com/open-zaak/open-zaak/issues/1834